### ISSUE TITLE
incorrect assignment of GameState.current_ship 

### ISSUE THREAD

**Issue by kwollw** [maxblan/socha-python-client#35]
## Description

current ship is not assigned correctly.

## Environment

-   Python version: 3.11.2
-   Operating System: Debian 12 
-   Version of the Python SoCha Client: 2.0.0

## Expected Behavior

current_ship team should remain the same whenever one team is requested for move.

## Actual Behavior

current_ship team switches every round

## Reproduction Steps (if applicable)

Run this client to observe this bug:
```
from socha import GameState, Move, Advance, Starter, IClientHandler

class Logic(IClientHandler):
    gameState: GameState

    def calculate_move(self) -> Move:
        print('calculate_move')
        print('-------------- current Ship ---------------------')
        print(self.gameState.current_ship)
        print('-------------------------------------------------')
        m = Move(actions = [Advance(1)])
        return m

    def on_update(self, state: GameState):
        self.gameState = state

if __name__ == "__main__":
  Starter(Logic())
```

**Comment by maxblan** [comment]
Thank you for the error report. Unfortunately I could not reproduce it. My output was always the correct team. Would it be possible that this has been fixed in 2.0.1?

```shell
2023-11-10 19:36:31,019: INFO - Joining game
2023-11-10 19:36:33,530: INFO - Ship's Team: TeamEnum.Two
2023-11-10 19:36:33,530: INFO - Sent Move(actions=[Advance(Advance { distance: 1 })]) after 0.0 seconds.
2023-11-10 19:36:33,667: INFO - Ship's Team: TeamEnum.Two
2023-11-10 19:36:33,668: INFO - Sent Move(actions=[Advance(Advance { distance: 1 })]) after 0.0 seconds.
2023-11-10 19:36:33,767: INFO - Ship's Team: TeamEnum.Two
2023-11-10 19:36:33,768: INFO - Sent Move(actions=[Advance(Advance { distance: 1 })]) after 0.0 seconds.
2023-11-10 19:36:33,835: INFO - Ship's Team: TeamEnum.Two
2023-11-10 19:36:33,835: INFO - Sent Move(actions=[Advance(Advance { distance: 1 })]) after 0.0 seconds.
2023-11-10 19:36:33,907: INFO - Ship's Team: TeamEnum.Two
2023-11-10 19:36:33,907: INFO - Sent Move(actions=[Advance(Advance { distance: 1 })]) after 0.0 seconds.
```

**Comment by kwollw** [comment]
I have investigated the problem in more detail. The error also occurs in version socha 2.0.3. It depends on the order in which the players are asked to move within a round.

Maybe this issue belongs to the server program. The server program used here is software-challenge-gui-24.1.5-linux.jar

### The following Program was tested:

```
from socha.api.networking.game_client import IClientHandler
from socha.starter import Starter
from socha import *
import logging

class Logic(IClientHandler):
    gameState: GameState

    def calculate_move(self) -> Move:
        logging.info(f"Ship's Team: {self.gameState.current_ship.team}")

        m = Move(actions = [Advance(1)])

        return m


    def on_update(self, state: GameState):
        self.gameState = state

if __name__ == "__main__":
  Starter(Logic())
```
### First Run:
Computer player as team 1, human as team 2
Round 1: (team 1 starts round 1)
                Computer player goes one space forward, human goes one space forward
Round 2: (**team 1 starts round 2**)
                Computer player moves one space forward, human moves one space forward
```
2023-11-18 20:29:25,009: INFO - You're running the latest version of socha (2.0.3)
2023-11-18 20:29:25,009: INFO - Joining game
2023-11-18 20:29:25,134: INFO - Ship's Team: TeamEnum.One
2023-11-18 20:29:25,134: INFO - Sent Move(actions=[Advance(Advance { distance: 1 })]) after 0.0 seconds.
2023-11-18 20:31:07,776: INFO - Ship's Team: TeamEnum.Two
2023-11-18 20:31:07,776: INFO - Sent Move(actions=[Advance(Advance { distance: 1 })]) after 0.0 seconds.
2023-11-18 20:31:14,135: INFO - Ship's Team: TeamEnum.One
2023-11-18 20:31:14,135: INFO - Sent Move(actions=[Advance(Advance { distance: 1 })]) after 0.0 seconds.
```

### Second Run:
Computer player as team 1, human as team 2
Round 1: (team 1 player starts round 1)
                Computer player goes one space forward, human player goes **two** spaces forward
Round 2: (**team 2 starts round 2**)
                human player moves one space forward, computer player moves one space forward

```
2023-11-18 20:46:01,846: INFO - You're running the latest version of socha (2.0.3)
2023-11-18 20:46:01,846: INFO - Joining game
2023-11-18 20:46:01,998: INFO - Ship's Team: TeamEnum.One
2023-11-18 20:46:01,999: INFO - Sent Move(actions=[Advance(Advance { distance: 1 })]) after 0.0 seconds.
2023-11-18 20:46:31,276: INFO - Ship's Team: TeamEnum.One
2023-11-18 20:46:31,276: INFO - Sent Move(actions=[Advance(Advance { distance: 1 })]) after 0.0 seconds.
```

**Comment by maxblan** [comment]
This should've been fixed with 82636ef9f022b1fac8502800e289e482061a1a2f resp. v2.0.4. I'll leave this issue open in case there is still a issue and close it in the next few days.

**Comment by kwollw** [comment]
We use an assertion to check the correctness of the gameState.current_ship attribute and could no longer detect any violations under socha 2.0.4.
